vm/src/stdlib/collections.rs

```diff
@@ -126,15 +126,31 @@ mod _collections {
             Ok(count)
         }
 
-        #[pymethod]
-        fn extend(&self, iter: PyIterable, vm: &VirtualMachine) -> PyResult<()> {
+        fn _extend(&self, iter: PyIterable, vm: &VirtualMachine) -> PyResult<()> {
             // TODO: use length_hint here and for extendleft
             for elem in iter.iter(vm)? {
                 self.append(elem?);
             }
             Ok(())
         }
 
+        #[pymethod]
+        fn extend(zelf: PyRef<Self>, iter: PyObjectRef, vm: &VirtualMachine) -> PyResult<()> {
+            // TODO: use length_hint here and for extendleft
+            if zelf.is(&iter) {
```

Comment:
Pretty sure you can follow what `list` does here. Extract the elements and then use `VecDeque`s `append`.

Reply:
I think he is trying somewhat better way. It will be `VecDeque` -> rust `Vec` -> `VecDeque` transition. In this case, it will only depending on iterators - which is more cpython way. (they call iternext and append for every iteration)

Reply:
Indeed, I'd argue there's precedent in the project for using `extract_elements` and it makes the function readable, plus, it *should* be relatively fast for the common case (`PyList`s). 

You've got more experience here so in the end of the day it's definitely your call.

Reply:
I agree that is faster in list. (for there, Vec -> Vec -> Vec)
I think our `extract_elements` is same as CPython's `PySequence_Fast_ITEMS`, right?
CPython uses PySequence_Fast_ITEMS for list, but not for deque. I think this is how also they handle the different situations.

Reply:
Oh, that sounds like another point.

Reply:
It should also be possible to special case `iter` being another `PyDeque` and just cloning its `VecDeque`.